lib/uuidUtils.js

```diff
@@ -0,0 +1,57 @@
+'use strict';
+
+const uuidRegEx = /^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}$/;
+const uuidV1Offset = 122192928000000000; //  100 nanosecond intervals since 15 October 1582
+
+const uuidUtils = {};
+
+/**
+ * Extract time in nanoseconds from UUID V1
+ *
+ * @param {string} uuidv1
+ * @return {Integer|null}
+ *
+ */
+uuidUtils.getTime = (uuidv1) => {
+    const uuidComponents = uuidv1.split('-');
+
+    return parseInt(
+        [
+            uuidComponents[2].substring(1),
+            uuidComponents[1],
+            uuidComponents[0]
+        ].join(''), 16);
+};
+
+/**
+ * Extract time in nanoseconds from UUID V1
+ *
+ * @param {string} uuidv1
+ * @return {Date|null}
+ *
+ */
+uuidUtils.getDate = (uuidv1) => {
+    if (!(typeof uuidv1 === 'string' || uuidv1 instanceof String)) {
+        return null;
+    }
+
+    // Drop the ns for conversion
+    return new Date(Math.floor((uuidUtils.getTime(uuidv1) - uuidV1Offset) / 10000));
```

Comment:
Suffers from same the problem as `getTime`. Once that is fixed, here you'll need only `new Date(uuidUtils.getTime(uuidv1))`. You'll also need to check for the return value, as `new Date(undefined)` returns the current date.

Reply:
Added check to return statement